pkg/updater/resultstore/resultstore.go

```diff
@@ -70,6 +73,117 @@ type processedResult struct {
 	TargetResults   map[string][]*singleActionResult
 }
 
+type timedError struct {
+	millis int64
+	err    error
+}
+type invocationGroup struct {
+	group       string
+	invocations []processedResult
+	errs        map[string]timedError
+}
+
+const (
+	// PartOf is the configuration value key which links to the group invocation ID.
+	PartOf = "EXECUTED_AS_PART_OF_INVOCATION"
+	// Originally indicates the associated non-overview indication.
+	Originally = "Original_test_invocation"
+)
+
+// buildIdentifier may assume its arguments are non-nil.
+type buildIdentifier func(configpb.TestGroup, processedResult) string
+
+var (
+	zero = time.Unix(0, 0)
+)
+
+func timeRunDate(milliseconds int64) time.Time {
+	return zero.Add(time.Duration(milliseconds) * time.Millisecond)
+}
+
+func invocationGroupID(inv processedResult, primary configpb.TestGroup_PrimaryGrouping, fallback configpb.TestGroup_FallbackGrouping, fallbackKey string, identifyBuild string) string {
```

Comment:
to identify groupId we just need test group configuration and processed result. better to use test group directly instead of having all of those individual fields to extract group ID

Reply:
we can also identify build using an already existing function which leverages a processed result and test group configuration

Reply:
> to identify groupId we just need test group configuration and processed result. better to use test group directly instead of having all of those individual fields to extract group ID

got it

Reply:
This depends imo. If we're using a lot of fields from the test group then I think it makes sense, but if we need only one or two, then just passing the fields often makes testing and understanding what the function depends on a lot easier.